pybossa/jobs.py

```diff
@@ -1358,6 +1359,33 @@ def check_and_send_task_notifications(project_id, conn=None):
         info = dict(project_name=project.name,
                     n_available_tasks=n_remaining_tasks)
         notify_task_progress(info, email_addr)
+
+        if webhook:
+            current_app.logger.info(u'Project {} the number of incomplete tasks: {}, \
+                                drops equal to or below target remaining: {}, hitting webhook url: {}'
+                                .format(project_id, n_remaining_tasks, target_remaining, webhook))
+            try:
+                headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
+                data = dict(project_id=project_id,
+                            project_name=project.name,
+                            remianing_tasks=n_remaining_tasks,
+                            target_remaining=target_remaining)
+                webhook_response = requests.post(webhook, data=json.dumps(data), headers=headers)
```

Comment:
upon success, webhook response with project details and data can be useful to log to troubleshoot if proj owner complains they did not receive notification.

Reply:
agree, will add logs

Reply:
Is it necessary to validate the response of Webhook? e.g. if it returns 200 we log success? in this case we may have to force user to provide response in their Webhook implementation

Reply:
i think every http request would return back some [response code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status).

Reply:
Yes, I mean if it returns some error code, should we log it and take some action, i.e. send email to owners? Or we don't care what it returns at all?

Reply:
isn't http responses 400-500+ an exception? if so, it would be handled by the new code you added in exception handler. So, i think it should be fine, but you can certainly confirm if needed.

Reply:
as discussed, the handing of response code need to be revisited